tests/unit/amo/components/TestReportUserAbuse.js

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react';
+
+import ReportUserAbuse, {
+  ReportUserAbuseBase,
+} from 'amo/components/ReportUserAbuse';
+import { setError } from 'core/actions/errors';
+import {
+  hideUserAbuseReportUI,
+  loadUserAbuseReport,
+  sendUserAbuseReport,
+  showUserAbuseReportUI,
+} from 'amo/reducers/userAbuseReports';
+import DismissibleTextForm from 'ui/components/DismissibleTextForm';
+import ErrorList from 'ui/components/ErrorList';
+import { dispatchClientMetadata } from 'tests/unit/amo/helpers';
+import {
+  createFakeUserAbuseReport,
+  createStubErrorHandler,
+  createUserAccountResponse,
+  fakeI18n,
+  shallowUntilTarget,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  const defaultRenderProps = {
+    errorHandler: createStubErrorHandler(),
+    i18n: fakeI18n(),
+    store: dispatchClientMetadata().store,
+    user: createUserAccountResponse({ id: 9 }),
+  };
+
+  function renderShallow({ ...props } = {}) {
+    return shallowUntilTarget(
+      <ReportUserAbuse {...defaultRenderProps} {...props} />,
+      ReportUserAbuseBase
+    );
+  }
+
+  it('renders a disabled button if no user exists', () => {
+    const root = renderShallow({ user: null });
+
+    expect(root.find('.ReportUserAbuse')).toHaveLength(1);
+    expect(root.find('.ReportUserAbuse-show-more'))
+      .toHaveProp('disabled', true);
+  });
+
+  it('renders a button to report an user', () => {
+    const root = renderShallow();
+
+    expect(root.find('.ReportUserAbuse')).toHaveLength(1);
+    expect(root.find('.ReportUserAbuse-show-more').prop('children'))
```

Comment:
Can you not use [`children()`](http://airbnb.io/enzyme/docs/api/ShallowWrapper/children.html)?

Reply:
Nice catch! I had to change the test to be `.toHaveText()` instead of `.toEqual()`, but that's nicer anyway. 👍